src/app/FakeLib/MSBuildHelper.fs

```diff
@@ -446,7 +464,10 @@ let MSBuildWithProjectProperties outputPath (targets : string) (properties : (st
             |> Set.unionMany
 
     let setBuildParam project projectParams =
-        { projectParams with Targets = targets |> split ';' |> List.filter ((<>) ""); Properties = projectParams.Properties @ properties project }
+        { projectParams with 
+            Targets = targets |> split ';' |> List.filter ((<>) "")
+            Properties = projectParams.Properties @ properties project
+            MaxCpuCount = Some ( Some (EnvironmentHelper.getMachineEnvironment().ProcessorCount)) }
```

Comment:
in fact, none of the `MSBuildXXXX` helpers set the core count, so they fall prey to this. The only way to get around it is to use raw `build setParams` yourself.